packages/core/src/integrations/mcp-server/index.ts

```diff
@@ -0,0 +1,48 @@
+import { fill } from '../../utils/object';
+import { wrapAllMCPHandlers } from './handlers';
+import { wrapTransportError, wrapTransportOnClose, wrapTransportOnMessage, wrapTransportSend } from './transport';
+import type { MCPServerInstance, MCPTransport } from './types';
+import { validateMcpServerInstance } from './validation';
+
+const wrappedMcpServerInstances = new WeakSet();
+
+/**
+ * Wraps a MCP Server instance from the `@modelcontextprotocol/sdk` package with Sentry instrumentation.
+ *
+ * Compatible with versions `^1.9.0` of the `@modelcontextprotocol/sdk` package.
+ */
+export function wrapMcpServerWithSentry<S extends object>(mcpServerInstance: S): S {
```

Comment:
Let's add an example usage snippet to the jsdoc here.

Reply:
done